src/main/java/app/daos/MovieDAO.java

```diff
@@ -116,8 +115,8 @@ public MovieDTO create(MovieDTO movieDTO) {
 
             foundCastMembers.forEach(castMember -> castMember.addMovie(movie));
 
-            movie.setCast(foundCastMembers);
-            movie.setGenres(getMovieGenres(movieDTO));
+            movie.setCast(new HashSet<>(foundCastMembers));
+            movie.setGenres(new HashSet<>(getMovieGenres(movieDTO)));
```

Comment:
Convert the `List<CastMember> foundCastMember` to a Set instead of converting it when calling `movie.setCast()`.
The same applies to `getMovieGenres`. Change the method to return a Set instead of converting it afterwards in `movie.setGenre()`

Reply:
Changed.